contracts/test/integration/utils/TimelockSimulation.sol

```diff
@@ -0,0 +1,96 @@
+pragma solidity =0.8.13;
+
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {DSTest} from "./../../unit/utils/DSTest.sol";
+import {Vm} from "./../../unit/utils/Vm.sol";
+
+import "hardhat/console.sol";
+
+contract TimelockSimulation is DSTest {
+    /// an array of actions makes up a proposal
+    struct action {
+        address target;
+        uint256 value;
+        bytes arguments;
+        string description;
+    }
+
+    function simulate(
+        action[] memory proposal,
+        TimelockController timelock,
+        address executor,
+        address proposer,
+        Vm vm
+    ) public {
+        uint256 delay = timelock.getMinDelay();
+        bytes32 salt = keccak256(abi.encode(proposal[0].description));
```

Comment:
Does this need to be checked for existence?

Reply:
It doesn't need to be checked for existence. If null, the salt would just be an empty string.